File: 66-plus-one/plus-one.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    let index = digits.length - 1;
   while(index !== 0 && digits[index]===9){
       digits[index] = 0;
       index--;
   } 
   if(digits.length === 1 && digits[index] === 9){
       digits = [1,0]
   } else if(index === 0 && digits[index] === 9){
       digits[index] = 0
       digits = [1,...digits]
   } else {
        digits[index] = digits[index] + 1;
   }
   return digits;
};

File: 66-plus-one/plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int i=digits.length-1;
        while(true)
        {
            if(i>0 && digits[i]==9)
            {
                digits[i]=0;
                if(i==0 && digits[i]==9)
                {
                    digits=new int[digits.length+1];
                    digits[0]=1;
                    break;
                }
                else 
                i=i-1;
            }
            else if(i==0 && digits[i]==9)
            {
                digits[0]=1;
                 digits=new int[digits.length+1];
                    digits[0]=1;
                    break;
            }
            else
            {
                digits[i]=digits[i]+1;                
                break;
            }
        }
        return digits;
        
    }
}